### PR TITLE
doc: Add FEC and Code Rate

### PR DESCRIPTION
**Pull request by pradeeka7**
#### Summary

Add learning material for Forward Error Correction and Code Rate (The Things Advanced Certification)

#### Screenshots

image

#### Changes

- ...
- ...

#### Notes for Reviewers

@benolayinka Can you review it?

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [X] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Review by benolayinka — APPROVED** [review]

File: doc/content/lorawan/fec-and-code-rate/_index.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+- 5/7
+- 4/8
+
+For example, if the code rate is 5/7, for every 5 bits of useful information, the coder generates a total of 7 bits of data, of which 2 bits are redundant.
+
+{{< figure src="fec-1.png" alt="" class="plain">}}
+
+Any Code Rate is usually applied with a spreading factor. For example, if you applied the code rate 4/5 with the Spreading Factor 7 _theoretically_ 7 x 4/5 = 5.6 bits carry useful information and the remaining 1.4 bits are used for forward error correction.
+
+Spreading factor 7 = the number of bits a symbol can encode is 7.
```
Comment (jpmeijers) [inline doc/content/lorawan/fec-and-code-rate/_index.md:25]: I'm not sure this is correct. With this argument SF12 should be faster, as you can fit 12 bits of info into a symbol. This is not true. SF refers to the time length of a symbol, not the number of bits per symbol. Do not mix the two. In other words, one can't link CR to SF.

Comment (pradeeka7) [reply doc/content/lorawan/fec-and-code-rate/_index.md:25]: @jpmeijers Thanks for your comment :) . As I know Spreading Factor is the number of (raw) bits that can be encoded by a symbol. Using a higher SF resulting a higher symbol duration (depends on the bandwidth).  Also, there are some other factors too: chip duration, chip rate, and symbol rate - all depend on the bandwidth and the SF used.

Fitting 12bits in an SF12 symbol doesn't make the data rate faster. Spreading Factor, Bandwidth, and Code Rate are the arguments for calculating the Data Rate.

Consider the following:
SF=7, BW= 125kHz, CR = 4/5
Data rate = 7 x  (125000/2^7) x (4/5) = ~5.5 kbits/s

SF=7, BW = 125kHz, CR = 4/6
Data rate = 7 x  (125000/2^7) x (4/6) = ~4.5 kbits/s

Read this thread:
https://electronics.stackexchange.com/questions/278192/understanding-the-relationship-between-lora-chips-chirps-symbols-and-bits

@benolayinka  Can you add this page for further review? If this is incorrect can you revert this PR?

Comment (jpmeijers) [reply doc/content/lorawan/fec-and-code-rate/_index.md:25]: Watch out here that you are not confusing `chip rate CR` with `coding rate CR`.

Comment (pradeeka7) [reply doc/content/lorawan/fec-and-code-rate/_index.md:25]: I have a very clear understanding of what Chip Rate and Code Rate are.
Chip Rate = Bandwidth in Hertz
For example, if the bandwidth is 125 kHz, the Chip Rate would be 125000 chips/s. If the Spreading Factor is 7, a symbol can hold 2^7 = 128 chips. 

Okay, I will remove the spreading factor section from this page and submit a new PR.

**Pull request merged**